homeassistant/components/alarm_control_panel/simplisafe.py

```diff
@@ -1,92 +1,50 @@
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
 
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_ALARM_ACTIVE = "alarm_active"
-ATTR_TEMPERATURE = "temperature"
-
-DATA_FILE = '.simplisafe'
-
-DEFAULT_NAME = 'SimpliSafe'
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_USERNAME): cv.string,
-    vol.Required(CONF_PASSWORD): cv.string,
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    vol.Optional(CONF_CODE): cv.string,
-})
+ATTR_ALARM_ACTIVE = 'alarm_active'
+ATTR_TEMPERATURE = 'temperature'
 
 
 async def async_setup_platform(
         hass, config, async_add_entities, discovery_info=None):
-    """Set up the SimpliSafe platform."""
-    from simplipy import API
-    from simplipy.errors import SimplipyError
-
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    name = config.get(CONF_NAME)
-    code = config.get(CONF_CODE)
-
-    websession = aiohttp_client.async_get_clientsession(hass)
-
-    config_data = await hass.async_add_executor_job(
-        load_json, hass.config.path(DATA_FILE))
-
-    try:
-        if config_data:
-            try:
-                simplisafe = await API.login_via_token(
-                    config_data['refresh_token'], websession)
-                _LOGGER.debug('Logging in with refresh token')
-            except SimplipyError:
-                _LOGGER.info('Refresh token expired; attempting credentials')
-                simplisafe = await API.login_via_credentials(
-                    username, password, websession)
-        else:
-            simplisafe = await API.login_via_credentials(
-                username, password, websession)
-            _LOGGER.debug('Logging in with credentials')
-    except SimplipyError as err:
-        _LOGGER.error("There was an error during setup: %s", err)
-        return
+    """Set up a SimpliSafe alarm control panel based on existing config."""
+    pass
 
-    config_data = {'refresh_token': simplisafe.refresh_token}
-    await hass.async_add_executor_job(
-        save_json, hass.config.path(DATA_FILE), config_data)
 
-    systems = await simplisafe.get_systems()
-    async_add_entities(
-        [SimpliSafeAlarm(system, name, code) for system in systems], True)
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up a SimpliSafe alarm control panel based on a config entry."""
+    systems = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
+    async_add_entities([
+        SimpliSafeAlarm(system, entry.data.get(CONF_CODE))
+        for system in systems
+    ], True)
 
 
 class SimpliSafeAlarm(AlarmControlPanel):
     """Representation of a SimpliSafe alarm."""
 
-    def __init__(self, system, name, code):
+    def __init__(self, system, code):
         """Initialize the SimpliSafe alarm."""
         self._attrs = {}
-        self._code = str(code) if code else None
-        self._name = name
+        self._code = code
+        self._dispatch_remove = None
```

Comment:
This doesn't need to be an instance attribute. It can be a local variable in `async_added_to_hass`.